### PR TITLE
OCT-1871: Home: Current GLM lock

### PR DESCRIPTION
**Pull request by jmikolajczyk**
## Description
Check after [PR-397](https://github.com/golemfoundation/octant/pull/397)

## Definition of Done

1. [ ] Acceptance criteria are met.
2. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
3. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
4. [ ] Unit tests are added unless there is a reason to omit them.
5. [ ] Automated tests are added when required.
6. [ ] The code is merged.
7. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
8. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlm/LockGlm.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+import { parseUnitsBigInt } from 'utils/parseUnitsBigInt';
+
+import styles from './LockGlm.module.scss';
+import LockGlmProps, { Step, OnReset } from './types';
+import { formInitialValues, validationSchema } from './utils';
+
+import LockGlmBudget from '../LockGlmBudget';
+import LockGlmNotification from '../LockGlmNotification';
+import LockGlmStepper from '../LockGlmStepper';
+import LockGlmTabs from '../LockGlmTabs';
```
Comment (aziolek) [inline client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlm/LockGlm.tsx:32]: Please use absolute paths.

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmBudget/LockGlmBudget.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useFormikContext } from 'formik';
+import React, { FC } from 'react';
+
+import { FormFields } from 'components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlm/types';
+
+import styles from './LockGlmBudget.module.scss';
+import LockGlmBudgetProps from './types';
+
+import LockGlmBudgetBox from '../LockGlmBudgetBox';
```
Comment (aziolek) [inline client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmBudget/LockGlmBudget.tsx:9]: Please use absolute paths.

File: client/src/components/Home/HomeGrid/HomeGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React, { memo, ReactNode } from 'react';
+
+import Grid from 'components/shared/Grid';
+
+import styles from './HomeGrid.module.scss';
+
+import HomeGridCurrentGlmLock from '../HomeGridCurrentGlmLock';
```
Comment (aziolek) [inline client/src/components/Home/HomeGrid/HomeGrid.tsx:7]: Please use absolute paths.

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabs/LockGlmTabs.tsx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+import useAvailableFundsGlm from 'hooks/helpers/useAvailableFundsGlm';
+import useDepositValue from 'hooks/queries/useDepositValue';
+import { formatUnitsBigInt } from 'utils/formatUnitsBigInt';
+import getFormattedGlmValue from 'utils/getFormattedGlmValue';
+import { parseUnitsBigInt } from 'utils/parseUnitsBigInt';
+
+import styles from './LockGlmTabs.module.scss';
+import LockGlmTabsProps from './types';
+
+import LockGlmTabsInputs from '../LockGlmTabsInputs';
```
Comment (aziolek) [inline client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabs/LockGlmTabs.tsx:19]: Please use absolute paths.

File: client/src/components/ui/DoubleValueSkeleton/types.ts [hunk 5]
```diff
@@ -1,3 +1,6 @@
+import { DoubleValueVariant } from '../DoubleValue/types';
```
Comment (aziolek) [inline client/src/components/ui/DoubleValueSkeleton/types.ts:1]: Please use absolute paths.

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmNotification/LockGlmNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import BoxRounded from 'components/ui/BoxRounded';
+import Button from 'components/ui/Button';
+import Svg from 'components/ui/Svg';
+import networkConfig from 'constants/networkConfig';
+import { arrowTopRight, checkMark, notificationIconWarning } from 'svg/misc';
+
+import styles from './LockGlmNotification.module.scss';
+import LockGlmNotificationProps from './types';
+
+const ButtonLinkWithIcon: FC<{ children?: React.ReactNode; transactionHash: string }> = ({
```
Comment (aziolek) [inline client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmNotification/LockGlmNotification.tsx:13]: We gotta extrat these types into `{...}Props` later, right?

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabsInputs/LockGlmTabsInputs.tsx [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+import {
+  comma,
+  floatNumberWithUpTo18DecimalPlaces,
+  floatNumberWithUpTo2DecimalPlaces,
+} from 'utils/regExp';
+
+import styles from './LockGlmTabsInputs.module.scss';
+import LockGlmTabsInputsProps from './types';
+
+const LockGlmTabsInputs = forwardRef<HTMLInputElement, LockGlmTabsInputsProps>(
```
Comment (aziolek) [inline client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabsInputs/LockGlmTabsInputs.tsx:16]: This component is a copy of `EarnGlmLockTabsInputs`. Do we keep the source, after it's already moved?

**Review by aziolek — CHANGES_REQUESTED** [review]

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmNotification/LockGlmNotification.tsx (hunk 6 above)
Comment (jmikolajczyk) [reply client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmNotification/LockGlmNotification.tsx:13]: Good catch. I extracted this.

File: client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabsInputs/LockGlmTabsInputs.tsx (hunk 7 above)
Comment (jmikolajczyk) [reply client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabsInputs/LockGlmTabsInputs.tsx:16]: I have a plan to remove every unused part of the app after finishing each view. Are you ok with it?

Comment (aziolek) [reply client/src/components/Home/HomeGridCurrentGlmLock/ModalLockGlm/LockGlmTabsInputs/LockGlmTabsInputs.tsx:16]: I would remove it already if it's not used, but we can keep it too.

**Review by aziolek — APPROVED** [review]

**Pull request merged**